torch/_guards.py

```diff
@@ -341,6 +344,22 @@ def current_frame(frame_summary):
         finally:
             tc.frame_summary_stack.pop()
 
+    @staticmethod
+    @contextlib.contextmanager
+    def current_loc(filename, lineno):
+        tc = TracingContext.get()
+        assert (
+            tc is not None
+        ), "Loc context manager must be called within an ongoing trace."
+        if len(tc.frame_summary_stack) > 0:
+            current_frame_name = tc.frame_summary_stack[-1].name
+        else:
+            current_frame_name = "<unknown>"
+        tc.loc_in_frame = traceback.FrameSummary(filename, lineno, current_frame_name)
+        try:
+            yield
+        finally:
+            tc.loc_in_frame = None
```

Comment:
I am a bit confused about the interaction between this and `frame_summary_stack`. Don't I want the current location in frame to be reported whenever I get the `frame_summary_stack`? I feel like the two of these should be interacting somehow.

Reply:
The current implementation of `frame_summary_stack` gives a stack of function definition locs rather than function call locs. (In other words, `def foo(x) at ...` rather than `foo(t) at ...`. This doesn't quite look like a traditional stack trace.

Reply:
OK, well, in a conventional stack trace, `def foo(x): ...` isn't included in a trace that throws midway through foo at all! So we probably don't want to do this?

Reply:
Also, maybe we should just be using https://github.com/pytorch/pytorch/pull/95541 ?

Reply:
OK, noting this as follow up work.